src/test/collectionHandlingFunctions.php

```diff
@@ -0,0 +1,304 @@
+    {
+        $yarnExample = ["name" => "Unicorn brand", 
+                        "shade" => "Rainbow", 
+                        "dominantColour" => "purple",
+                        "composition" => "Cashmere",
+                        "yarnType" => "Super Chunky", 
+                        "lengthInMeters" => 100,
+                        "skeinNumber" => 2, 
+                        "project"=> "Cool rainbow hat",
+                        "image" => "blue_scarf.jpg"      
+                    ];
+        $expectedOutput = '<div class="collection_item">';
+        $expectedOutput .= '<h3>Unicorn brand Rainbow</h3>';
+        $expectedOutput .= '<img src="images/blue_scarf.jpg" alt="A picture of Unicorn brand in shade Rainbow">';              
+        $expectedOutput .= '<h4>Description</h4>';
+        $expectedOutput .= '<ul>';
+        $expectedOutput .= '<li>Dominant colour: purple</li>';
+        $expectedOutput .= '<li>Yarn type: Super Chunky</li>';
+        $expectedOutput .= '<li>Composition: Cashmere</li>';
+        $expectedOutput .= '<li>Length of a skein: 100 m</li>';            
+        $expectedOutput .= '<li>Quantity: 2 skeins in stash</li>';
+        $expectedOutput .= '<li>Allocated to project: Cool rainbow hat</li>';
+        $expectedOutput .= '</ul>';
+        $expectedOutput .= '</div>'; 
+        
+        $actualOutput = describeCollectionItem($yarnExample);
+
+        $this->assertEquals($expectedOutput, $actualOutput);
+    }
+
+ //The following functions check that the output is correct when one of the fields is missing 
+ // that item in the database    
+    public function testSuccessDescribeCollectionItem_undefinedShade()
+    {
+        $yarnExample = ["name" => "Unicorn brand", 
+                        "shade" => NULL, 
+                        "dominantColour" => "purple",
+                        "composition" => "Cashmere",
+                        "yarnType" => "Super Chunky", 
+                        "lengthInMeters" => 100,
+                        "skeinNumber" => 2, 
+                        "project"=> "Cool rainbow hat",
+                        "image" => "blue_scarf.jpg"      
+                    ];
+        $expectedOutput = '<div class="collection_item">';
+        $expectedOutput .= '<h3>Unicorn brand</h3>';
+        $expectedOutput .= '<img src="images/blue_scarf.jpg" alt="A picture of Unicorn brand">';              
+        $expectedOutput .= '<h4>Description</h4>';
+        $expectedOutput .= '<ul>';
+        $expectedOutput .= '<li>Dominant colour: purple</li>';
+        $expectedOutput .= '<li>Yarn type: Super Chunky</li>';
+        $expectedOutput .= '<li>Composition: Cashmere</li>';
+        $expectedOutput .= '<li>Length of a skein: 100 m</li>';            
+        $expectedOutput .= '<li>Quantity: 2 skeins in stash</li>';
+        $expectedOutput .= '<li>Allocated to project: Cool rainbow hat</li>';
+        $expectedOutput .= '</ul>';
+        $expectedOutput .= '</div>'; 
```

Comment:
I think if you have this in one public function, you don't need to repeat this code? you can just use the expectedOutcome variable as it already has this value stored in it?

Reply:
also as discussed, ignore!

Reply:
I added a few comments to make it clearer